pkg/operator/controller/controller_cabundle_configmap.go

```diff
@@ -0,0 +1,193 @@
+				_, _, err = r.currentCABundleConfigMap(destName)
+				return err
+			}
+		}
+	}
+
+	// remove ca bundle configmaps that are not referred in dns anymore.
+	cmSelector, err := metav1.LabelSelectorAsSelector(&cmLabelSelector)
+	if err != nil {
+		return err
+	}
+	cmListOpts := []client.ListOption{
+		client.MatchingLabelsSelector{
+			Selector: cmSelector,
+		},
+		client.InNamespace("openshift-dns"),
+	}
+	var cmList corev1.ConfigMapList
+	if err := r.cache.List(context.TODO(), &cmList, cmListOpts...); err != nil {
+		return fmt.Errorf("failed to list ca bundle configmaps: %w", err)
+	}
+	for _, cm := range cmList.Items {
+		referredInDNS := false
+		for _, name := range configmapNames {
+			destName := CABundleConfigMapName(name)
+			if cm.Name == destName.Name {
+				referredInDNS = true
+				break
+			}
+		}
+		if !referredInDNS {
+			if err := r.client.Delete(context.TODO(), &cm); err != nil {
+				if !errors.IsNotFound(err) {
+					return fmt.Errorf("failed to delete configmap: %w", err)
+				}
+			} else {
+				logrus.Infof("deleted configmap %s/%s", cm.Namespace, cm.Name)
+			}
+		}
+	}
+
+	return nil
+}
+
+// desiredCABundleConfigMap returns the desired CA bundle configmap.  Returns a
+// Boolean indicating whether a configmap is desired, as well as the configmap
+// if one is desired.
+func desiredCABundleConfigMap(dns *operatorv1.DNS, haveSource bool, sourceConfigmap *corev1.ConfigMap, name types.NamespacedName) (bool, *corev1.ConfigMap, error) {
+	if !haveSource {
+		return false, nil, nil
+	}
+	if dns.DeletionTimestamp != nil {
+		return false, nil, nil
+	}
+	cm := corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name.Name,
+			Namespace: name.Namespace,
+			Labels:    cmLabels,
+		},
+		Data: sourceConfigmap.Data,
+	}
+	cm.SetOwnerReferences([]metav1.OwnerReference{dnsOwnerRef(dns)})
+
+	return true, &cm, nil
+}
+
+// currentCABundleConfigMap returns the current configmap.  Returns a Boolean
+// indicating whether the configmap existed, the configmap if it did exist, and
+// an error value.
+func (r *reconciler) currentCABundleConfigMap(name types.NamespacedName) (bool, *corev1.ConfigMap, error) {
+	if len(name.Name) == 0 {
+		return false, nil, nil
+	}
+	cm := &corev1.ConfigMap{}
+	if err := r.client.Get(context.TODO(), name, cm); err != nil {
+		if errors.IsNotFound(err) {
+			return false, nil, nil
+		}
+		return false, nil, err
+	}
+	return true, cm, nil
+}
+
+// updateCABundleConfigMap updates a configmap.  Returns a Boolean indicating
+// whether the configmap was updated, and an error value.
+func (r *reconciler) updateCABundleConfigMap(current, desired *corev1.ConfigMap) (bool, error) {
+	if caBundleConfigmapsEqual(current, desired) {
+		return false, nil
+	}
+	updated := current.DeepCopy()
+	updated.Data = desired.Data
+	if err := r.client.Update(context.TODO(), updated); err != nil {
+		return false, err
+	}
+	logrus.Infof("updated configmap %s/%s", updated.Namespace, updated.Name)
```

Comment:
`ensureCABundleConfigMaps` also logs when an update has been done.  One of these log statements can be deleted.

Reply:
Deleted this one.